dnsmasq: conflicts with odhcpd-ipv6only

candrews opened the pull request:
dnsmasq-full and dnsmasq-dhcpv6 conflict with odhcpd-ipv6only.

Having two dhcpv6 servers installed at the same time is not a good idea.

See also https://github.com/openwrt/luci/issues/2147

Thanks for your contribution to OpenWrt!

To help keep the codebase consistent and readable,
and to help people review your contribution,
we ask you to follow the rules you find in the wiki at this link
https://openwrt.org/submitting-patches

Please remove this message before posting the pull request.

candrews committed:
dnsmasq: conflict with odhcpd-ipv6only

dnsmasq-full and dnsmasq-dhcpv6 conflict with odhcpd-ipv6only.

Having two dhcpv6 servers installed at the same time is not a good idea.

See also https://github.com/openwrt/luci/issues/2147

Signed-off-by: Craig Andrews <candrews@integralblue.com>

ynezz commented:
Please can you kindly read https://openwrt.org/submitting-patches one more time, and fix your commit accordingly? Thanks!

candrews commented:
> Please can you kindly read https://openwrt.org/submitting-patches one more time, and fix your commit accordingly? Thanks!

I re-read that and adjusted the commit:
* I realized the sign off was missing, so I added it.
* In the commit subject, s/conflicts/conflict/ which I think is imperative in the desired style

I'm happy to make any other changes, but I _think_ I'm in compliance with the guidelines now :crossed_fingers:

PussAzuki commented:
I have used this combination ( dnsmasq-full + odhcpd-ipv6only ) for a while without any strange things happening . I am using a 6in4 network and it seems that odhcpd-ipv6only will become a DHCPv6 server by default, and dnsmasq-full's DHCPv6 function will be disabled.I hope that you can check carefully the conflicts between them in all network states.

dedeckeh commented:
I understand the reasoning behind this PR however it fails to cover the case if odhcpd (iso odhcpd-ipv6only)  is selected as two DHCPv6 servers will be installed as well with on top the odhcpd DHCPv4 server

PussAzuki commented:
According to my test , this combination dnsmasq/dnsmasq-full + odhcpd/odhcpd-ipv6only can perform dhcpv4 role in dnsmasq, odhcpd works normally when performing dhcpv6 role .But I did not think about this usage dnsmasq-full/dnsmasq-dhcpv6 to perform the role of dhcpv6 and odhcpd to perform the role of dhcpv4 is not acceptable according to what you said,. I was excited, sorry.😔

peterwillcn commented:
@dedeckeh @ynezz  @PussAzuki  I think it is time to redesign the relationship before dnsmasq/dnsmasq-full + odhcpd/odhcpd-ipv6only. I think proper combination of dnsmasq/dnsmasq-full and odhcpd/odhcpd-ipv6only functions is crucial to the stability of the system. It is recommended to remove odhcpd-ipv6only and dnsmasq. By default, use dnsmasq-full and odhcpd. LuCI adds options to enable some functions.

PussAzuki commented:
By default, when any combination of dnsmasq-full/dnsmasq+odhcpd/odhcpd-ipv6only is installed in the system, the former will only serve as the dns server and dhcp server, and the latter will only serve as the dhcpv6 server. If this is the case, there will be no errors, and I personally think that the two dhcp and dhcpv6 servers are not installed in the system at the same time. In theory, if one of the two problems occurs with dhcp/dhcpv6, you can replace the configuration. And submit a bug instead of waiting and doning nothing. If you have to ask if I have tested all the above combinations? I certainly tested it by coincidence

jow- commented:
We want to allow both packages to coexist, so this PR won't be taken.
